modules/ROOT/pages/build-an-https-service.adoc

```diff
@@ -0,0 +1,146 @@
+Learn how to design and develop an API using our xref:general::api-led-overview.adoc[Build an API from Start to Finish] tutorial.
+
+== Configure HTTPS Service
+
+Modify your service to HTTPS to deploy your app to CloudHub:
+
+. Generate a `keystore.jks` file using the JDK `keytool` utility on the command line. You must also specify the hostname on the command line to generate a self-signed certificate. +
+
++
+For example, the following command with hostname `SAN=DNS:localhost,IP:127.0.0.1` creates a `keystore.jks` file:
++
+[source,text,linenums]
+----
+keytool -genkeypair -keystore keystore.jks   -dname "CN=localhost, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown"  -keypass password  -storepass password  -keyalg RSA  -sigalg SHA1withRSA  -keysize 2048  -alias mule  -ext SAN=DNS:localhost,IP:127.0.0.1 -validity 9999
+----
+
+. Add the generated `keystore.jks` file to your Anypoint Studio Mule app folder in `src/main/resources`.
+. Configure the HTTP Listener in your Mule app, specifying:
+
+* Any host IP addresses you want to use with the HTTPS scheme for the value of `host`
+* The `${https.port} variable for the value of `port`
+
+[source,xml,linenums]
+----
+...
+<http:listener-config name="HTTPS_Listener_Configuration" protocol="HTTPS" host="0.0.0.0" port="${https.port}">
+     <tls:context>
+         <tls:key-store path="keystore.jks" keyPassword="${keystore.password}" password="${password}"/>
+     </tls:context>
+ </http:listener-config>
+...
+----
+
+[start=4]
+. Include a `config-ref` reference to the HTTPS global listener configuration.
+
+[source,xml,linenums]
+----
+<flow name="httpsserviceFlow">
+    <http:listener config-ref="HTTPS_Listener_Configuration" path="hello"/>
+</flow>
+----
+
+[start=5]
+. Configure the HTTP Requester using your required TLS configuration to enable HTTPS requests to external addresses:
++
+[source,xml,linenums]
+----
+...
+<http:request-config name="HTTP_Request_Configuration" protocol="HTTPS" host="0.0.0.0" port="${https.port}" >
+     <tls:context>
+         <tls:key-store path="keystore.jks" password="${password}" keyPassword="${keystore.password}" />
+     </tls:context>
+</http:request-config>
+...
+----
+
+[start=6]
+. Include a `config-ref` reference to the HTTPS global request configuration:
++
+[source,xml,linenums]
+----
+<flow name="httpsserviceFlow">
+  <http:request config-ref="HTTP_Request_Configuration" path="some-path" method="GET" host="0.0.0.0" port="${https.port}"/>
+</flow>
+----
+
+[start=7]
+. Your application is now ready to be deployed on CloudHub. You can access your endpoint using the HTTPS address: for example, `+https://yourdomain.cloudhub.io+`
+
+== Configure Services under API Manager Proxies
+
+If you are prompted to download a proxy from API Manager and need to configure it for HTTPS, follow the steps in the HTTP Connector configuration for the HTTP Requester. The HTTP Listener configuration is provided as a template.
```

Comment:
Follow the steps from...where? Is there a task that should be named or xref'd here? Or do you mean "use the same steps that you used in Anypoint Connector for HTTP to configure the HTTP Requestor?

Reply:
Also...is this a prerequisite to the steps that follow?